packages/slate/src/models/node.js

```diff
@@ -1370,17 +1378,55 @@ class Node {
    */
 
   getPath(key) {
-    let child = this.assertNode(key)
-    const ancestors = this.getAncestors(key)
-    const path = []
+    if (!this.hasNode(key)) {
+      throw new Error(`Could not find a node with key "${key}".`)
+    }
+    const path = this.getPathAsString(key)
+    if (path.length === 0) {
+      return []
+    }
+    return path.split(' ').map(x => parseInt(x, 10))
+  }
 
-    ancestors.reverse().forEach(ancestor => {
-      const index = ancestor.nodes.indexOf(child)
-      path.unshift(index)
-      child = ancestor
-    })
+  /**
+   * Get the path of a descendant node by `key`, as string
+   * PERF: concat string is much faster than concat Array (at least in v8),
+   * becuase no extra space is required in v8's ConString
+   *
+   * @param {String|Node} key
+   * @returns {Array}
+   */
 
-    return path
+  getPathAsString(key) {
+    key = assertKey(key)
+    if (this.key === key) return ''
+    if (this.nodes.size === 0) return null
+    let result = null
+    // PERF: often the node keys is generated with sequences;
```

Comment:
After users edited the document, this assumption might be wrong.

Before:

```diff
[
  '1',
  '2'
]
```

And then, user inserts a new line:

```diff
  [
    '1',
+   '3',
    '2'
  ]
```

Reply:
@benjycui Yes, I agree the assumption can be wrong.  But it is like a costless try that "If the guess is right, then it is good; If the guess is wrong, it does not make things worse".  

If we have no idea about users behavior, `findLast` is not expected worse than `find`.  In this case, this methods is worse the previous one only by two read keys (which seems to be costless). 

If key is not a number string, the key is NaN, then `Math.abs(lastKey - searchKey) < Math.abs(searchKey - firstKey)` is false, then the method will be findIndex anyway~

Reply:
Perhaps I need to add an annotation to elaborate that~